docs/vectorizer/document-vectorization.md

```diff
@@ -0,0 +1,516 @@
+
+```
+AWS_ACCESS_KEY_ID=your_access_key
+AWS_SECRET_ACCESS_KEY=your_secret_key
+```
+
+The user must have appropriate S3 read permissions for the buckets containing your documents.
+
+
+**2. Assume Role-based Authentication**
+
+You can also use the `aws_role_arn` parameter to assume an IAM role. This is what Timescale Cloud uses, but it is usually not necessary if you self-host since the worker will run in the same AWS environment as your buckets:
+
+```sql
+SELECT ai.create_vectorizer(
+    'document'::regclass,
+    loading => ai.loading_uri(
+        column_name => 'uri',
+        aws_role_arn => 'arn:aws:iam::123456789012:role/S3AccessRole'
+    ),
+    -- other configuration...
+);
+```
+
+The role must have appropriate S3 read permissions for the buckets containing your documents.
+
+### Setup for Timescale-cloud-hosted pgai
+
+For Timescale Cloud installations, only role-based authentication via `assume_role_arn` is supported.
+
+#### Create a Role for s3 access
+First you need to create a role that Timescale can assume:
+
+```bash
+aws iam create-role \
+  --role-name timescale-vectorizer-s3-access \
+  --assume-role-policy-document '{
+    "Version": "2012-10-17",
+    "Statement": [
+      {
+        "Effect": "Allow",
+        "Principal": {
+          "AWS": "arn:aws:iam::142548018081:role/timescale-pgai-vectorizer"
+        },
+        "Action": "sts:AssumeRole",
+        "Condition": {
+          "StringLike": {
+            "sts:ExternalId": "projectId/serviceId"
+          }
+        }
+      }
+    ]
+  }'
+```
+
+Note that the assumeRole permission needs you to replace the `projectId/serviceId` with the actual project and service id of your Timescale Cloud installation. You can find this in the Timescale Cloud console. This is a security measure that prevents the [confused deputy problem](https://docs.aws.amazon.com/IAM/latest/UserGuide/confused-deputy.html), which would otherwise allow other Timescale Cloud users to access your buckets if they guessed your role name and accountId.
+
+#### Grant Permissions to your bucket to the role
+
+```bash
+aws iam put-role-policy \
+  --role-name timescale-vectorizer-s3-access \
+  --policy-name S3AccessPolicy \
+  --policy-document '{
+    "Version": "2012-10-17",
+    "Statement": [
+      {
+        "Action": [
+          "s3:GetObject"
+        ],
+        "Effect": "Allow",
+        "Resource": [
+          "arn:aws:s3:::test",
+          "arn:aws:s3:::test/*"
+        ]
+      }
+    ]
+  }'
+```
+
+#### Get the role ARN
+```bash
+aws iam get-role --role-name timescale-s3-role-test --query 'Role.Arn' --output text
+```
+#### Configure it in your ai.loading_uri:
+
+```sql
+ai.loading_uri(
+    column_name => 'uri',
+    aws_role_arn => 'arn:aws:iam::123456789012:role/timescale-vectorizer-s3-access'
+)
+```
+
+
+### Syncing S3 to a Documents Table
```

Comment:
I have the feeling this belongs more to like an example or tutorial rather than to this guide.